services/workflows-service/src/workflow/workflow.service.ts

```diff
@@ -645,18 +663,69 @@ export class WorkflowService {
     const updatedWorkflow = await this.updateContextById(workflowId, {
       documents: documentsWithDecision,
     });
-
     const correlationId = await this.getCorrelationIdFromWorkflow(updatedWorkflow);
+    const entityId = updatedWorkflow.businessId || updatedWorkflow.endUserId;
 
     this.workflowEventEmitter.emit('workflow.context.changed', {
-      oldRuntimeData: runtimeData,
+      oldRuntimeData: workflow,
       updatedRuntimeData: updatedWorkflow,
       state: updatedWorkflow.state,
-      entityId: (updatedWorkflow.businessId || updatedWorkflow.endUserId) as string,
+      entityId,
       correlationId: correlationId,
     });
 
-    return updatedWorkflow;
+    // TODO: Check against `contextSchema` or a policy if the length of documents is equal to the number of tasks defined.
+    const resolvedStates = ['approved', 'rejected'];
+    const shouldResolve =
+      workflowDefinition?.config?.completedWhenTasksResolved && workflow?.status === 'active';
+    const allDocumentsResolved =
+      shouldResolve &&
+      safeEvery(
+        updatedWorkflow?.context?.documents,
+        (document: DefaultContextSchema['documents'][number]) => {
+          if (!document?.decision?.status) return false;
+
+          return resolvedStates.includes(document?.decision?.status);
+        },
+      );
+
+    const workflowService = createWorkflow({
+      runtimeId: workflow?.id,
+      definition: workflowDefinition?.definition,
+      definitionType: workflowDefinition?.definitionType,
+      workflowContext: {
+        machineContext: workflow?.context,
+        state: workflow?.state ?? workflowDefinition?.definition?.initial,
+      },
+    });
+
+    const snapshot = workflowService.getSnapshot();
+    const isFinal = workflowDefinition?.definition?.states?.[snapshot?.value]?.type === 'final';
+    const isResolved =
+      isFinal || allDocumentsResolved || workflow?.status === WorkflowRuntimeDataStatus.completed;
+
+    const updatedWorkflowWithResolve = await this.workflowRuntimeDataRepository.updateById(
+      workflow?.id,
+      {
+        data: {
+          status: allDocumentsResolved ? 'completed' : workflow?.status,
+          resolvedAt: isResolved ? new Date().toISOString() : null,
+        },
+      },
+    );
+
+    if (isResolved) {
```

Comment:
Sending a webhook for `workflow.completed` was previously missing from this endpoint.